api/v1alpha1/common_types.go

```diff
@@ -86,6 +93,33 @@ const (
 	Injected Phase = "Injected"
 )
 
+type RecordEvent struct {
+	Type      RecordEventType  `json:"type"`
+	Stage     RecordEventStage `json:"stage"`
+	Message   string           `json:"message,omitempty"`
+	Timestamp *metav1.Time     `json:"timestamp"`
+}
+
+type RecordEventType string
+
+const (
+	// Succeed means the stage of this event is successful
+	Succeed RecordEventType = "Succeed"
+	// Failed means the stage of this event is failed
+	Failed RecordEventType = "Failed"
+)
+
+type RecordEventStage string
+
+const (
+	// Injection means this event is recorded, when we inject the chaos
+	// typically, when we call impl.Apply()
+	Injection RecordEventStage = "Injection"
+	// Recovery means this event is recorded, when we recover the chaos
+	// typically, when we call impl.Recover()
+	Recovery RecordEventStage = "Recovery"
```

Comment:
Maybe we could use the words "Apply" and "Recover" (instead of "Injection" and "Recovery" )as stage names for reducing thinking?